pages/api/payments/paymentId/index.ts

```diff
@@ -0,0 +1,28 @@
+import { Decimal } from '@prisma/client/runtime/library'
+import { generatePaymentId } from 'services/transactionService'
+import { parseAddress } from 'utils/validators'
+import { RESPONSE_MESSAGES } from 'constants/index'
+
+export default async (req: any, res: any): Promise<void> => {
+  if (req.method === 'GET') {
```

Comment:
This should work the same way as it does now, but should actually be a POST request. It will create an entry in the DB (more than one, if the address doesn't exist), it's not idempotent and it changes state.